lib/stagehand/staging/checklist.rb

```diff
@@ -18,10 +18,12 @@ def self.related_entries(entries, relation_filter = nil)
 
         entries_to_spider = Array.wrap(entries)
         while entries_to_spider.present?
-          contained_matching = CommitEntry.contained.matching(entries_to_spider).to_a
-          contained_matching.select! {|entry| relation_filter.call(entry) } if relation_filter.present?
+          contained_matching = CommitEntry.contained.matching(entries_to_spider)
+          if relation_filter.present?
```

Comment:
Do we need to check `present?`, won't it be nil/non-nil? Same below with `association_filter`.